google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/model.scala

```diff
@@ -77,66 +77,171 @@ object PubSubConfig {
-
-  /**
-   * Java API: create [[PubSubMessage]]
-   */
-  def create(data: String) =
-    PubSubMessage(data)
+  def create(data: Option[String],
+             attributes: Option[immutable.Map[String, String]],
+             messageId: String,
+             publishTime: Instant) = new PubSubMessage(data, attributes, messageId, publishTime)
 
-  @deprecated("Setting messageId when creating message for publishing is futile.", "1.1.0")
-  def create(data: String, messageId: String) =
-    PubSubMessage(data, messageId)
 }
 
-final case class PublishRequest(messages: immutable.Seq[PubSubMessage])
+final class PublishRequest private (val messages: immutable.Seq[PublishMessage]) {
+
+  override def equals(other: Any): Boolean = other match {
+    case that: PublishRequest => messages == that.messages
+    case _ => false
+  }
+
+  override def hashCode: Int = messages.hashCode
+
+  override def toString: String = "PublishRequest(" + messages.mkString("[", ",", "]") + ")"
+}
 
 object PublishRequest {
-  def of(messages: java.util.List[PubSubMessage]): PublishRequest =
-    PublishRequest(messages.asScala.toList)
+
+  def apply(messages: immutable.Seq[PublishMessage]): PublishRequest = new PublishRequest(messages)
+
+  def of(messages: java.util.List[PublishMessage]): PublishRequest =
+    new PublishRequest(messages.asScala.toList)
 }
 
 /**
  * A message as it is received
  * @param ackId acknowledgement id. This id is used to tell pub/sub the message has been processed.
  * @param message the pubsub message including its data.
  */
-final case class ReceivedMessage(ackId: String, message: PubSubMessage)
+final class ReceivedMessage private (val ackId: String, val message: PubSubMessage) {
+
+  override def equals(other: Any): Boolean = other match {
+    case that: ReceivedMessage => ackId == that.ackId && message == that.message
+    case _ => false
+  }
+
+  override def hashCode: Int = java.util.Objects.hash(ackId, message)
+
+  override def toString: String = "ReceivedMessage(ackId=" + ackId.toString + ",message=" + message.toString + ")"
+}
+
+object ReceivedMessage {
+
+  def apply(ackId: String, message: PubSubMessage): ReceivedMessage =
+    new ReceivedMessage(ackId, message)
 
-final case class AcknowledgeRequest(ackIds: immutable.Seq[String])
+  def create(ackId: String, message: PubSubMessage): ReceivedMessage =
+    new ReceivedMessage(ackId, message)
+}
+
+final class AcknowledgeRequest private (val ackIds: immutable.Seq[String]) {
+
+  override def equals(other: Any): Boolean = other match {
+    case that: AcknowledgeRequest => ackIds == that.ackIds
+    case _ => false
+  }
+
+  override def hashCode: Int = ackIds.hashCode
+
+  override def toString: String = "AcknowledgeRequest(" + ackIds.mkString("[", ",", "]") + ")"
+}
 
 object AcknowledgeRequest {
+
+  def apply(ackIds: String*): AcknowledgeRequest =
+    new AcknowledgeRequest(ackIds.toList)
+
+  def apply(ackIds: immutable.Seq[String]): AcknowledgeRequest =
```

Comment:
This does not compile with Scala 2.13 anymore:
```
[error] /home/travis/build/akka/alpakka/google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/model.scala:206:7: double definition:
[error] def apply(ackIds: String*): akka.stream.alpakka.googlecloud.pubsub.AcknowledgeRequest at line 203 and
[error] def apply(ackIds: Seq[String]): akka.stream.alpakka.googlecloud.pubsub.AcknowledgeRequest at line 206
[error] have same type after erasure: (ackIds: Seq)akka.stream.alpakka.googlecloud.pubsub.AcknowledgeRequest
[error]   def apply(ackIds: immutable.Seq[String]): AcknowledgeRequest =
[error]       ^
```

because since Scala 2.13 varargs are encoded with `immutable.Seq`. Lets only keep the varargs method as it is possible to pass in a list to varargs method with a `: _*` notation.

Reply:
:+1:
Sorry, I didn't check build state. I'll make it green today.

Reply:
"Today" took me 3 days, sorry. Build is green now.